ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -379,20 +379,25 @@ boolean isCaughtUp() {
   }
 
   void shutdown() throws InterruptedException, StoreException, IOException {
-    if (enabled.get()) {
-      logger.info("Hard delete shutdown initiated");
-      enabled.set(false);
-      hardDeleteLock.lock();
-      try {
-        pauseCondition.signal();
-      } finally {
-        hardDeleteLock.unlock();
+    long startTimeInMs = time.milliseconds();
+    try {
+      if (enabled.get()) {
+        logger.info("Hard delete shutdown initiated");
+        enabled.set(false);
+        hardDeleteLock.lock();
+        try {
+          pauseCondition.signal();
+        } finally {
+          hardDeleteLock.unlock();
+        }
+        shutdownLatch.await();
+        logger.info("Hard delete shutdown complete");
+        throttler.close();
+        pruneHardDeleteRecoveryRange();
```

Comment:
do you guys this we need a metric to track time for pruning and persisting the token ?